domino/domino.py

```diff
@@ -1047,6 +1047,50 @@ def datasets_remove(self, dataset_ids: list):
             url = self._routes.datasets_details(dataset_id)
             self.request_manager.delete(url)
 
+    def datasets_upload_file(
+        self,
+        dataset_id: str,
+        local_path_to_file_or_directory: str,
+        file_upload_setting: str = None,
+        max_workers: int = None,
+        target_chunk_size: int = None,
+        target_relative_path: str = None
+    ) -> str:
+        """Upload file to dataset with multithreaded support.
+
+        Args:
+            dataset_id: id of dataset whose rw snapshot the file will be uploaded to
+            local_path_to_file_or_directory: path to file in local machine
+            file_upload_setting: setting to resolve naming conflict, one of Ignore, Rename, Overwrite (default)
+            max_workers: max amount of threads (default: 10)
+            target_chunk_size: max chunk size for multipart upload (default: 8MB)
+            target_relative_path: path in the dataset to upload the file to
+        Returns local path to uploaded file
+        """
+        uploader = datasets.Uploader(
+            csrf_no_check_header=self._csrf_no_check_header,
+            dataset_id=dataset_id,
+            local_path_to_file_or_directory=local_path_to_file_or_directory,
+            log=self.log,
+            request_manager=self.request_manager,
+            routes=self._routes,
+
+            file_upload_setting=file_upload_setting,
+            max_workers=max_workers,
+            target_chunk_size=target_chunk_size,
+            target_relative_path=target_relative_path
+        )
+
+        try:
+            uploader.start_upload_session()
```

Comment:
I feel like you could easily make this a context manager https://www.pythonmorsels.com/creating-a-context-manager/ to avoid having the user of this class know which method to call first (since there are no arguments once created)